subsys: Bluetooth: LLL: Add functions that allow configure Radio for reception and sampling of CTE

ppryga-nordic opened the pull request:
The PR includes:
- fix for build error from direction finding compilation when `IS_ENABLE(CONFIG_BT_CTLR_DF_ADV_CTE_TX)`  is `false`
- API to configure Radio to receive and sample CTE

Part of former code responsible for configuration of CTE Tx was refactored to comply with code responsible for CTE Rx configuration.

ppryga-nordic committed:
Bluetooth: controller: radio: Add API to configure CTE Rx

Add Radio configuration API that gives possiblity configure
reception and sampling of CTE.
Part of alrady implemented API was re-factored to comply
with new API and do complete initialization of registers:
DFECTRL1 and CTEINLINECONF.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

kruithofa commented on subsys/bluetooth/controller/ll_sw/ull_df.c:
```diff
@@ -31,6 +31,7 @@
 #define LOG_MODULE_NAME bt_ctlr_ull_df
 #include "common/log.h"
 
+#if IS_ENABLED(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
```
For consistency use '#if defined' instead of '#if IS_ENABLED'
See also PR #32609

ppryga replied:
Corrected. I'll create a separate PR to clean this in all DF related files (if IS_ENABLES is still available)

ppryga commented:
Added change requested by @kruithofa.
Rebased onto master.

ppryga commented:
Removed cte_length argument from functions: `radio_df_cte_rx_2us_switching`, `radio_df_cte_rx_4us_switching`.
`cte_length` is not used when CTEINLINE is enabled. In such situation CTE length is send in `CTEInfo` filed in PDU.

ppryga commented:
Fixed compliance issue

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_df.c:
```diff
@@ -46,6 +47,7 @@
 
 static struct lll_df_adv_cfg lll_df_adv_cfg_pool[CONFIG_BT_CTLR_ADV_AUX_SET];
 static void *df_adv_cfg_free;
+#endif /* #if CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```
```suggestion
#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```

ppryga replied:
Corrected

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_df.c:
```diff
@@ -54,13 +56,15 @@ static void *df_adv_cfg_free;
  * @return Pointer to lll_df_adv_cfg or NULL if there is no more free memory.
  */
 static struct lll_df_adv_cfg *df_adv_cfg_acquire(void);
+#endif /* #if CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```
```suggestion
#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```

ppryga replied:
Corrected

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_df.c:
```diff
@@ -96,11 +100,12 @@ int ull_df_reset(void)
 		 sizeof(lll_df_adv_cfg_pool) / sizeof(struct lll_df_adv_cfg),
 		 &df_adv_cfg_free);
-
+#endif /* #if CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```
```suggestion
#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
```

ppryga replied:
Corrected

cvinayak reviewed (changes requested):
nitpicks, otherwise looks good.

ppryga-nordic committed:
tests: Bluetooth: df: Fix undefinded symbols error

Remove undefined functions (unsued anymore).
Add missing stub implementation for new DF related radio functions.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga commented:
Added one commit with:
- removed undefined symbols error
- added missing symbols to radio stubs implementation.

@cvinayak your requested modifications are implemented.

carlescufi commented:
@cvinayak @thoh-ot @kruithofa please take another look

cvinayak reviewed (approved):
LGTM.

kruithofa reviewed (approved):
LGTM
